### ISSUE TITLE
Disappears when clicking show desktops.

### ISSUE THREAD

**Issue by rockmanvnx6** [Twometer/NoFences#2]
When clicking on show desktop (bottom right corner) all the Fences disappear.

**Comment by Twometer** [comment]
I don't think it is possible with WinForms to prevent Show Desktop from hiding the fences, as it does not minimize the window but bring the desktop to foreground.
However, once you reopen any window, they should reappear.

**Comment by rockmanvnx6** [comment]
Understandable. Would you recommend any workaround?

**Comment by Twometer** [comment]
I am trying to get a fix to work, but as for now you have to bring up any app after clicking show desktop. After that, all fences will reappear.

**Comment by Lucarnosky** [comment]
I think this should Help: https://stackoverflow.com/questions/938924/get-the-minimize-box-click-of-a-wpf-window/938956#938956

**Comment by Twometer** [comment]
@Lucarnosky Unfortunately, that doesn't work. The app uses WinForms, but I think there is also a way to get the Minimize Box Event. 
The main problem is that a (winforms?) app does not get any event for when the show desktop button is clicked, as it does not minimize the windows...

**Comment by leonelsr** [comment]
Hello! I've just found and started using NoFences, and it's just great!

I do think this issue is solved by now, isn't it?

Well, if it isn't, I've had the "CTRL+D minimization issue" before on personal use projects, and the snippet found on this Stack Overflow question did the trick: https://stackoverflow.com/questions/35045060/how-to-keep-window-visible-at-all-times-but-not-force-it-to-be-on-top

I hope it helps, please take a look!

Thanks!
(I might clone NoFences and try to make it un-minimizable myself, later today or this week... I'll turn anything I do with it into one or more PRs)

**Comment by Twometer** [comment]
Hi! Thank you for the suggestion.
I have actually tried implementing a similar solution in `Win32/DesktopUtil.cs` which however did not prevent Win+D. I've just tried the code from StackOverflow as well, but for me it still disappears when pressing Windows+D
If you get it to work, I would greatly appreciate the PR 😄

**Comment by leonelsr** [comment]
I'm on lunch time, but I've already found the solution I've used on my repositories (though I have no time right now to try it on NoFences).

This is the piece of code that did the trick:
```CSharp
        private const Int32 GWL_STYLE = -16;
        private const Int32 WS_MAXIMIZEBOX = 0x00010000;
        private const Int32 WS_MINIMIZEBOX = 0x00020000;
// ...
            m_hwnd = new WindowInteropHelper(this).Handle;

            Int32 windowStyle = GetWindowLongPtr(m_hwnd, GWL_STYLE);
            SetWindowLongPtr(m_hwnd, GWL_STYLE, windowStyle & ~WS_MAXIMIZEBOX & ~WS_MINIMIZEBOX);
```

You may try it right away if you have time before me!

**Comment by Twometer** [comment]
Even with this code in place, for me it still disappears with Win+D... Maybe you have more luck than I do :-)
Looking forward to your contributions, sounds great!

**Comment by MahmoudKhaiti** [comment]
woah... woah... woah

how to I enable it even? I'm new to Github

**Comment by Twometer** [comment]
@MahmoudKhaiti what do you mean by enabling it?

**Comment by GordnCZ** [comment]
Hi. Just noticed this repo while finding free alternative.
I've been already experimenting with this kind of issues.
I already have some code ready for this so I might contribute by fixing this stuff.
But it's really dirty solution.

BTW: I just made my account. I previously deleted my GH account when I moved all my stuff to NAS...

**Comment by Twometer** [comment]
@GordnCZ Sounds nice, I'd be glad if you find a way to fix it. I tried leonelsr's code already when he posted it, but for me it did not work.

**Comment by GordnCZ** [comment]
@Twometer I currently don't have time. But I'll try to look at it on the weekend.

**Comment by GordnCZ** [comment]
@Twometer Hi. Sorry for taking so long. Didn't have time at all.
Anyway I found a solution.

It's pretty easy. All you need to do is to replace line 46 in DesktopUtil.cs with this:
`SetWindowLongPtr(handle, GWL_HWNDPARENT, nWinHandle.ToInt32());`

And uncomment line 60 in FencesWindow.cs.
Even background blur works 😉

**Comment by Twometer** [comment]
Thanks everyone for the suggestions! Fixed in #14
